### PR TITLE
Create workflow.yml

### PR DESCRIPTION
**Pull request by NkwaTambe**
Create a GitHub Actions workflow to automate the build process for a React application. The workflow should trigger on pushes to the main branch and pull requests targeting the main branch. It should set up the Node.js environment, install dependencies, build the React application, and archive the build artifacts. This workflow will streamline the continuous integration process, ensuring that the React application is built consistently and ready for deployment.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
```
Comment (stephane-segning) [inline .github/workflows/node.js.yml:19]: Why're you building for multiple versions?

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```
Comment (stephane-segning) [inline .github/workflows/node.js.yml:10]: Why're not building for all branches?

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+    - run: npm ci
+    - run: npm run build --if-present
+    - run: npm test
```
Comment (stephane-segning) [inline .github/workflows/node.js.yml:31]: Where is this `test` coming from?

**Review by stephane-segning — CHANGES_REQUESTED** [review]
This project is not using npm but yarn. So review your work again.

**Review by stephane-segning — APPROVED** [review]

File: .github/workflows/node.js.yml (hunk 3 above)
Comment (NkwaTambe) [reply .github/workflows/node.js.yml]: fixed

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (NkwaTambe) [reply .github/workflows/node.js.yml]: fixed sir

**Review by Ngha-Boris — APPROVED** [review]
Good

**Pull request merged**